Add config for disable validation in specific presenters

kopecmi8 opened the pull request:
Following previous [pull request](https://github.com/Kdyby/HtmlValidatorPanel/pull/9) I changed my solution and now you can specify presenters where you dont want to run validation in config file.

For example it can be done like this
```javascript
htmlValidator: 
	disableValidationInPresenters: [
		Company\Fe\CustomPresenter
	]
```
Its allow you to specify presenters its outputs you don't want to validate with HTML validator.

Commit:
Add config for disable validation in specific presenters

simPod commented on ValidatorPanel.php:
```diff
@@ -858,33 +866,45 @@ public function stopBuffering()
+		
+		$validationEnabled = TRUE;
+		
+		foreach($this->disableValidationInPresenters as $disableValidationInPresenter){
```
Are you sure about this? Personally, I'd check whether `$application->getPresenter()` is in `$this->disableValidationInPresenters` and if it is so, I'd exit early. I believe it would look like this:

```php
if(in_array(get_class($application->getPresenter()), $disableValidationInPresenter, true)){
    return;
}
```

Then we can completely throw away `$validationEnabled` I guess.

kopecmi8 replied:
Yes it´s easier way how to do this. Only one thing was wrong, you have to call ob_end_flush() first before you can return.

simPod commented on ValidatorPanel.php:
```diff
@@ -769,6 +769,11 @@ class ValidatorPanel implements Tracy\IBarPanel
 	private $errors = array();
+	
+	/**
+	 * @var array
```
I haven't setup any project to test this yet so I'm not sure what data type will be here. I guess it can be changed to `@var string[]`

kopecmi8 replied:
You are right. I fix this.

simPod commented on ValidatorPanel.php:
```diff
@@ -769,6 +769,11 @@ class ValidatorPanel implements Tracy\IBarPanel
+	/**
+	 * @var array
+	 */
+	private $disableValidationInPresenters = array();
```
`$presentersToDisableValidationFor` sounds better in 🇬🇧

kopecmi8 replied:
Yes it sound better i changed this name

simPod reviewed (changes requested):
Please update docs as well

kopecmi8 commented:
I made new commit. I corrected things you suggested and made the documentation. Other thing I added support for config parameter to be a string if you have only one presenter where you want to disable validation in. 
And I want to ask, if  I have to squash the commits into one?

simPod commented on ValidatorPanel.php:
```diff
@@ -865,46 +867,40 @@ public function stopBuffering()
 
 	/**
 	 * Validate.
+	 * @param \Nette\Application\Application $application
```
this parameter doc is redundant

simPod commented:
> I added support for config parameter to be a string

I can't find it but also I think it's not necessary, it's better not to mix types

simPod commented:
BTW the changes look good and IMO the final diff is now clean 👍

kopecmi8 commented on DI/ValidatorExtension.php:
```diff
@@ -29,9 +29,14 @@ public function loadConfiguration()
+		
+		$presentersToDisableValidationFor = array();
+		if(isset($this->config['presentersToDisableValidationFor'])) {
+			$presentersToDisableValidationFor = is_array($this->config['presentersToDisableValidationFor']) ? $this->config['presentersToDisableValidationFor'] : array($this->config['presentersToDisableValidationFor']);
```
There is condition which determinate if config parameter is single string and convert it to array. If you think is unnecessary I can change it like this:

```php
if(isset($this->config['presentersToDisableValidationFor']) && is_array($this->config['presentersToDisableValidationFor'])) {
  $presentersToDisableValidationFor =$this->config['presentersToDisableValidationFor'];
}
```

simPod replied:
Ah I see. Yup, I'd drop it, the array is just fine and the complexity is lower then

kopecmi8 commented:
I made those changes and squashed commits into the single one. I think now is code prepared to be merged

simPod commented on DI/ValidatorExtension.php:
```diff
@@ -29,9 +29,14 @@ public function loadConfiguration()
+		
 		$builder->addDefinition($this->prefix('panel'))
-			->setClass('Kdyby\Extension\Diagnostics\HtmlValidator\ValidatorPanel')
+			->setClass('Kdyby\Extension\Diagnostics\HtmlValidator\ValidatorPanel', Array($presentersToDisableValidationFor) )
```
one last thing, we can do `$this->config['presentersToDisableValidationFor'] ?? []` and remove the whole block above

Meanwhile, I'd increase min PHP v to something like 7.1, would it work for you?

kopecmi8 replied:
This is little bit problem for me, because I use this extension on project where is PHP 5.6
Can you make this changes in another version in the future?

simPod replied:
I was afraid so. In that case lets change it to:

```php
$presentersToDisableValidationFor = isset($this->config['presentersToDisableValidationFor']) ? $this->config['presentersToDisableValidationFor'] : [];
```

What do you think?

kopecmi8 replied:
I made this change.

simPod commented on ValidatorPanel.php:
```diff
@@ -858,11 +868,16 @@ public function stopBuffering()
 	 * Validate.
 	 */
-	public function validate()
+	public function validate(\Nette\Application\Application $application)
```
is the Application being passed here? I don't see it in https://github.com/Kdyby/HtmlValidatorPanel/blob/51df775e068a0c378c2ee00a59093d10d5089e24/DI/ValidatorExtension.php#L39

simPod replied:
I see 👍

simPod reviewed (approved):
LGTM 🚢 

Thanks!
